explorer-ui/src/components/Activities/Activities.jsx

```diff
@@ -0,0 +1,154 @@
+// This function calls API methods by fetch function (you can use XMLHttpRequest or $.ajax instead):
+import ActivityGrid from "../ActivityGrid/ActivityGrid";
+import { useState, useEffect } from "react";
+import ActivityCard from "../ActivityCard/ActivityCard";
+import "./Activities.css"
+//You should get your API key at https://opentripmap.io
+const apiKey = "5ae2e3f221c38a28845f05b66afc7a4b942f1b2a702f9c54e864e3c6";
+  // Init global variables for paging:
+
+  const pageLength = 10; // number of objects per page
+
+  let lon; // place longitude
+  let lat; // place latitude
+
+  let offset = 0; // offset from first object in the list
+  let count; // total objects count
+
+
+export default function Activities({country}) {
+  const [eventData, setEventData] = useState([]);
+  const [xidList, setXIDList] = useState([]);
+  const [events, setEvents] = useState([]);
+
+// Fetch data from the open trips maps API
+  function apiGet(method, query) {
+    return new Promise(function (resolve, reject) {
+      var otmAPI =
+        "https://api.opentripmap.com/0.1/en/places/" +
+        method +
+        "?apikey=" +
+        apiKey;
+      if (query !== undefined) {
+        otmAPI += "&" + query;
+      }
+      fetch(otmAPI)
+        .then((response) => response.json())
+        .then((data) => resolve(data))
+        .catch(function (err) {
+          console.log("Fetch Error :-S", err);
+        });
+    });
+  }
+
+  // This block uses the placename from input textbox and gets place location from API. If place was found it calls list loading function:
+
+  function handleTripSubmit(event) {
+    event.preventDefault();
+    console.log("clicked");
+    let name = "Moscow";
+    apiGet("geoname", "name=" + name).then(function (data) {
+      let message = "Name not found";
+      if (data.status == "OK") {
+        message = data.name + ", " + "Moscow";
+        lon = 105.3188;
+        lat = 61.524;
+        firstLoad();
+      }
+      document.getElementById("info").innerHTML = `${message}`;
```

Comment:
hmm so in general prefer using react vs browser APIs i.e call the fxn from component -> set state with result

Reply:
fixed!